test/image/mocks/sort_by_value_matching_axes.json

```diff
@@ -0,0 +1,43 @@
+{
+  "data": [{
+    "type": "scatter",
+    "orientation": "v",
+    "x": ["a", "b", "c"],
+    "y": [7, 2, 3],
+
+    "z": [
+      [7, 2, 3],
+      [0, 0, 0],
+      [0, 0, 0]
+    ],
+    "dimensions": [{
+      "label": "DimensionA",
+      "values": ["a", "b", "c"]
+    }, {
+      "label": "DimensionB",
+      "values": [7, 2, 3]
+    }]
+  }, {
+    "type": "bar",
+    "x": ["a", "b", "c"],
+    "y": [10, 20, 30],
```

Comment:
Hmm. I don't understand why the categories show up as `[b, c, a]`. Shouldn't the `y` values for the bar trace by considered also?

Reply:
> Hmm. I don't understand why the categories show up as `[b, c, a]`. Shouldn't the `y` values for the bar trace by considered also?

At the moment, not if they are on a different axis: only the traces associated with the axis to be sorted are accounted for. In this mock, the `bar` trace is on different axes namely `(x|y)axis2`. It ends inheriting the order of categories of `xaxis` because it `matches` it but that's all. I thought this behavior would be useful.

We could account for the values of all traces across all matching axes IF we also support a way to specify traces to ignore as suggested @nicolaskruchten (see https://github.com/plotly/plotly.js/issues/3606#issuecomment-492694202) so that we can reproduce the current baseline.

Reply:
> not if they are on a different axis: only the traces associated with the axis to be sorted are accounted for. In this mock, the `bar` trace is on different axes namely `(x|y)axis2`. It ends inheriting the order of categories of `xaxis` because it `matches` it but that's all. I thought this behavior would be useful.

Thanks for the explanation. That makes perfect sense :+1: